rerun_cpp/src/rerun/demo_utils.cpp

```diff
@@ -0,0 +1,87 @@
+#include "demo_utils.hpp"
+#include <algorithm>
+
+namespace rerun {
+    namespace demo {
+
+        std::vector<components::Position3D> grid(
+            components::Position3D from, components::Position3D to, size_t n
+        ) {
+            std::vector<components::Position3D> output;
+            for (float z : linspace(from.z(), to.z(), n)) {
+                for (float y : linspace(from.y(), to.y(), n)) {
+                    for (float x : linspace(from.x(), to.x(), n)) {
+                        output.push_back({x, y, z});
+                    }
+                }
+            }
+            return output;
+        }
+
+        void color_spiral(
+            size_t num_points, float radius, float angular_step, float angular_offset, float z_step,
+            std::vector<components::Position3D>& out_points,
+            std::vector<components::Color>& out_colors
+        ) {
+            out_points.reserve(num_points);
+            out_colors.reserve(num_points);
+
+            for (size_t i = 0; i < num_points; ++i) {
+                float angle = static_cast<float>(i) * angular_step * TAU + angular_offset;
+                out_points.push_back({
+                    sinf(angle) * radius,
+                    cosf(angle) * radius,
+                    static_cast<float>(i) * z_step,
+                });
+
+                out_colors.push_back(colormap_turbo_srgb(static_cast<float>(i) / num_points));
+            }
+        }
+
+        template <size_t N>
+        static float dot(const float (&a)[N], const float (&b)[N]) {
+            float sum = 0.0f;
```

Comment:
nitpick, but it is always a good idea to use higher precision accumulators

```suggestion
            double sum = 0.0f;
```